Islandora-Devops/ansible-role-fcrepo
# SQL-based Fedora object persistence (Issue 1244)

**GitHub Issue**: /Islandora-CLAW/CLAW/issues/1244

# What does this Pull Request do?

Adds options to use the MySQL or Postgresql databases for Fedora object stores instead of just the simple-file configuration (still the default)

# What's new?
* Added
    - fcrepo_persistence variable to select which persistence configuration should be used
    - variables to setup the database: fcrepo_db_name, fcrepo_db_user, and fcrepo_db_password
    - task files for MySQL and Postgresql
* Removed the repository.json config file (we now download the appropriate config based on the fcrepo_persistence variable
* Does this change require documentation to be updated? Just this README (updated)
* Does this change add any new dependencies? No.
* Does this change require any other modifications to be made to the repository
 (ie. Regeneration activity, etc.)? Only if you want to migrate your existing Fedora to a SQL-based object persistence.
* Could this change impact execution of existing code? No.

# How should this be tested?

- Update your claw-playbook requirements.yml:
```
- src: https://github.com/seth-shaw-unlv/ansible-role-fcrepo
  name: Islandora-Devops.fcrepo
  version: issue-1244
```
- Vagrant up

That should give you the file-based object persistence as before. The README gives information about using the role with a SQL-based object persistence. 

**Alternatively**, you can use /Islandora-Devops/claw-playbook/pull/132 to test provisioning with a MySQL store.

# Additional Notes:

I've tested simple-file and MySQL; not with Postgresql.

# Interested parties
@dannylamb, @whikloj, @Islandora-Devops/committers
